### PR TITLE
Move `OptimismHardfork` to new crate `reth_optimism_forks`

### LINKED ISSUE #10518: Move `OptimismHardfork` to `reth-optimism-chainspec`
### Describe the feature

Move `reth_ethereum_forks::OptimismHardfork` into new crate `reth-optimism-chainspec`

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by emhane**
Closes https://github.com/paradigmxyz/reth/issues/10518

- Adds crate `reth-optimism-forks`
- Moves `reth_ethereum_forks::OptimismHardfork` and `reth_ethereum_forks::OptimismHardforks` to `reth-optimism-forks`

### CODE REVIEW

**Review by mattsse — COMMENTED** [review]
that makes sense, but I wonder if this breaks --dev mode for op

https://github.com/paradigmxyz/reth/blob/0746479e3c873e0f83c533871598dd1d749d0041/crates/chainspec/src/spec.rs#L108-L115

**Comment by emhane** [comment]
> that makes sense, but I wonder if this breaks --dev mode for op
> 
> https://github.com/paradigmxyz/reth/blob/0746479e3c873e0f83c533871598dd1d749d0041/crates/chainspec/src/spec.rs#L108-L115

let me try :)

**Comment by emhane** [comment]
re-request review @mattsse

File: .github/workflows/unit.yml
Comment (mattsse) [inline .github/workflows/unit.yml:1]: why do we need to change this?

**Review by mattsse — COMMENTED** [review]
cool, incremental progress.

one q re CI change, otherwise lgtm

File: .github/workflows/unit.yml
Comment (emhane) [reply .github/workflows/unit.yml:1]: because of the `parse_dev` test

Comment (mattsse) [reply .github/workflows/unit.yml:1]: could we remove that, not worth changing the ci for and adding a single expectation in ci

Comment (emhane) [reply .github/workflows/unit.yml:1]: then I have to remove the whole test

**Review by mattsse — APPROVED** [review]

**Pull request merged**